DEVELOPING.md

```diff
@@ -8,6 +8,16 @@ Before running through the release it's good to run the build and the tests loca
 also test-drive the commit in an existing Bazel workspace to sanity check functionality.
 
 #### Steps
+1. (Optional) modify the `./python/pip_install/tools/requirements.txt` file and run `./tools/update_pip_deps.sh` script:
+   1. Using system python:
+   ```bash
+   ./tools/update_pip_deps.py
+   ```
+   1. Using bazel toolchain:
+   ```bash
+   bazel run --enable_bzlmod @python_3_10_9//:python3 -- ./tools/update_pip_deps.py
```

Comment:
Ah, er, hm. Out of scope for this PR, but, isn't 3.10.9 both correct _and_ incorrect?

IIRC, the tools get run with the Python version from e.g. `pip.parse(python_version=...)` (and similar for workspace builds).

So it's correct if someone is using 3.10, but not _really_ correct for someone using e.g. 3.9.

Please tell me I'm wrong? :)

Otherwise -- ugh gah >.<